api/semaphore.go

```diff
@@ -460,8 +484,24 @@ func (s *Semaphore) monitorLock(session string, stopCh chan struct{}) {
 	kv := s.c.KV()
 	opts := &QueryOptions{RequireConsistent: true}
 WAIT:
+	retries := s.opts.MonitorRetries
+RETRY:
 	pairs, meta, err := kv.List(s.opts.Prefix, opts)
 	if err != nil {
+		// TODO (slackpad) - Make a real error type here instead of using
+		// a string check.
+		const serverError = "Unexpected response code: 500"
```

Comment:
Maybe at least move this out of the retry loop? Maybe we could use like an `IsServerError(error)` since I think we do this in a bunch of other places too IIRC.